src/humanloop/eval_utils/run.py

```diff
@@ -0,0 +1,739 @@
+    # TODO: Update the API so this is not necessary
+    evaluator_stats_by_path = _get_evaluator_stats_by_path(
+        stat=next(
+            (stat for stat in stats.run_stats if stat.run_id == run_id),
+            None,  # type: ignore
+        ),
+        evaluation=evaluation,
+    )
+    if evaluator_path in evaluator_stats_by_path:
+        evaluator_stat = evaluator_stats_by_path[evaluator_path]
+        score = _get_score_from_evaluator_stat(stat=evaluator_stat)
+        if score >= threshold:  # type: ignore
+            logger.info(
+                f"{GREEN}✅ Latest eval [{score}] above threshold [{threshold}] for evaluator {evaluator_path}.{RESET}"
+            )
+            return True
+        else:
+            logger.info(
+                f"{RED}❌ Latest score [{score}] below the threshold [{threshold}] for evaluator {evaluator_path}.{RESET}"
+            )
+            return False
+    else:
+        raise ValueError(f"Evaluator {evaluator_path} not found in the stats.")
+
+
+def _check_evaluation_improvement(
+    evaluation: EvaluationResponse,
+    evaluator_path: str,
+    stats: EvaluationStats,
+    run_id: str,
+) -> Tuple[bool, float, float]:
+    """
+    Check the latest version has improved across for a specific Evaluator.
+
+    :returns: A tuple of (improvement, latest_score, delta since previous score)
+    """
+    # TODO: Update the API so this is not necessary
+
+    latest_evaluator_stats_by_path = _get_evaluator_stats_by_path(
+        stat=next(
+            (stat for stat in stats.run_stats if stat.run_id == run_id),
+            None,  # type: ignore
+        ),
+        evaluation=evaluation,
+    )
+    if len(stats.run_stats) == 1:
+        logger.info(f"{YELLOW}⚠️ No previous versions to compare with.{RESET}")
+        return True, 0, 0
+
+    previous_evaluator_stats_by_path = _get_evaluator_stats_by_path(stat=stats.run_stats[-2], evaluation=evaluation)
+    if evaluator_path in latest_evaluator_stats_by_path and evaluator_path in previous_evaluator_stats_by_path:
+        latest_evaluator_stat = latest_evaluator_stats_by_path[evaluator_path]
+        previous_evaluator_stat = previous_evaluator_stats_by_path[evaluator_path]
+        latest_score = _get_score_from_evaluator_stat(stat=latest_evaluator_stat)
+        previous_score = _get_score_from_evaluator_stat(stat=previous_evaluator_stat)
+        diff = round(latest_score - previous_score, 2)  # type: ignore
+        if diff >= 0:
+            logger.info(f"{CYAN}Change of [{diff}] for Evaluator {evaluator_path}{RESET}")
+            return True, latest_score, diff  # type: ignore
+        else:
+            logger.info(f"{CYAN}Change of [{diff}] for Evaluator {evaluator_path}{RESET}")
+            return False, latest_score, diff  # type: ignore
+    else:
+        raise ValueError(f"Evaluator {evaluator_path} not found in the stats.")
+
+
+def _add_log_to_evaluation(
```

Comment:
bit of a strange name of the function.

This is encapsulating the logic of doing external evaluator logs only.

It's not adding log to evaluator

Reply:
Renamed to _run_local_evaluators